compiler/tests/fixtures/demos/uniswap.fe

```diff
@@ -0,0 +1,365 @@
+
+    reserve0: u256
+    reserve1: u256
+    block_timestamp_last: u256
+
+    price0_cumulative_last: u256
+    price1_cumulative_last: u256
+    k_last: u256
+
+    event Approval:
+        idx owner: address
+        idx spender: address
+        value: u256
+
+    event Transfer:
+        idx from: address
+        idx to: address
+        value: u256
+
+    event Mint:
+        idx sender: address
+        amount0: u256
+        amount1: u256
+
+    event Burn:
+        idx sender: address
+        amount0: u256
+        amount1: u256
+        idx to: address
+
+    event Swap:
+        idx sender: address
+        amount0_in: u256
+        amount1_in: u256
+        amount0_out: u256
+        amount1_out: u256
+        idx to: address
+
+    event Sync:
+        reserve0: u256
+        reserve1: u256
+
+    pub def __init__():
+        self.factory = msg.sender
+
+    pub def factory() -> address:
+        return self.factory
+
+    pub def token0() -> address:
+        return self.token0
+
+    pub def token1() -> address:
+        return self.token1
+
+    def _mint(to: address, value: u256):
+        self.total_supply = self.total_supply + value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(address(0), to, value)
+
+    def _burn(from: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.total_supply = self.total_supply - value
+        emit Transfer(from, address(0), value)
+
+    def _approve(owner: address, spender: address, value: u256):
+        self.allowances[owner][spender] = value
+        emit Approval(owner, spender, value)
+
+    def _transfer(from: address, to: address, value: u256):
+        self.balances[from] = self.balances[from] - value
+        self.balances[to] = self.balances[to] + value
+        emit Transfer(from, to, value)
+
+    pub def approve(spender: address, value: u256) -> bool:
+        self._approve(msg.sender, spender, value)
+        return true
+
+    pub def transfer(to: address, value: u256) -> bool:
+        self._transfer(msg.sender, to, value)
+        return true
+
+    pub def transferFrom(from: address, to: address, value: u256) -> bool:
+        assert self.allowances[from][msg.sender] >= value
+
+        self.allowances[from][msg.sender] = self.allowances[from][msg.sender] - value
+        self._transfer(from, to, value)
+        return true
+
+    pub def balanceOf(account: address) -> u256:
+        return self.balances[account]
+
+    # TODO: support structs in the public interface (https://github.com/ethereum/fe/issues/282)
+    pub def get_reserves(): # -> ThreeNums:
+#        return ThreeNums(
+#            num1=self.reserve0,
+#            num2=self.reserve1,
+#            num3=self.block_time_stamp
+#        )
+        pass
+
+    # called once by the factory at time of deployment
+    pub def initialize(token0: address, token1: address):
+        assert msg.sender == self.factory, "UniswapV2: FORBIDDEN"
```

Comment:
If I'm not mistaken, passing a string to assert such as `"UniswapV2: FORBIDDEN"` is recognized by the parser but doesn't have any effect yet, right? If so, we should put a comment and issue on it as well.

Reply:
yeah, good point